src/components/suggest/suggest_item.tsx

```diff
@@ -1,9 +1,55 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { FunctionComponent } from 'react';
+import { CommonProps } from '../common';
 import classNames from 'classnames';
-import { EuiIcon, IconPropType } from '../icon';
+import { EuiIcon } from '../icon';
 
-const colorToClassNameMap = {
+interface Type {
+  iconType: string;
+  color: string | keyof typeof colorToClassNameMap;
+}
+
+export type EuiSuggestItemProps = CommonProps & {
+  /**
+   * Takes 'iconType' for EuiIcon and 'color'. 'color' can be tint1 through tint9.
+   */
+  type: Type;
+
+  /**
+   * Label or primary text.
+   */
+  label: string;
+
+  /**
+   * Description or secondary text (optional).
+   */
+  description?: string;
+
+  /**
+   * Label display is 'fixed' by default. Label will increase its width beyond 50% if needed with 'expand'.
+   */
+  labelDisplay?: keyof typeof labelDisplayToClassMap;
+
+  /**
+   * Handler for click on a suggestItem.
+   */
+  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
```

Comment:
A little shorter and nicer representation,

```suggestion
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
```